src/mbgl/util/mapbox.cpp

```diff
@@ -6,53 +6,80 @@ namespace mbgl {
 namespace util {
 namespace mapbox {
 
-const std::string mapbox = "mapbox://";
+const std::string protocol = "mapbox://";
+const std::string baseURL = "https://api.mapbox.com/";
+const std::string draft = "draft";
+
+struct MapboxURLParts {
+    std::string endpoint;
+    std::string user;
+    std::string id;
+    bool draft;
+};
+
+bool isMapboxURL(const std::string& url) {
+    return url.compare(0, protocol.length(), protocol) == 0;
+}
+
+void parseMapboxURL(const std::string& url, MapboxURLParts &parts) {
```

Comment:
This function should return its `MapboxURLParts` result. C++ will optimize away the copy.